### PR TITLE
[Enhancement] Support mix upsert and delete for primary key table

### PR DESCRIPTION
**Pull request by sevev**
Signed-off-by: zhangqiang <qiangzh95@gmail.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

PrimaryKey table support delete operation in data import right now. However, if we mixed upsert and delete in one data import, we only support a small batch data(smaller than one segment file size). If the imported data size need to flush more than one segment, we will refuse the import request.

This pr will support mixed upsert and delete data import for PrimaryKey table and don't need to limit the segment num.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by decster — DISMISSED** [review]

**Comment by sevev** [comment]
run starrocks_admit_test

File: be/src/storage/rowset/rowset_writer.cpp [hunk 1]
```diff
@@ -404,8 +408,13 @@ Status HorizontalRowsetWriter::flush_chunk(const Chunk& chunk, SegmentPB* seg_in
...
     case FlushChunkState::UPSERT:
         break;
+    case FlushChunkState::DELETE:
+        _flush_chunk_state = FlushChunkState::MIXED;
+        break;
+    case FlushChunkState::MIXED:
+        break;
     default:
-        return Status::Cancelled(_dump_mixed_segment_delfile_not_supported());
+        return Status::Cancelled("UNKNOWN flush chunk state");
```
Comment (chaoyli) [inline be/src/storage/rowset/rowset_writer.cpp:417]: 1. Add a tablet info in this error Status.

Comment (sevev) [reply be/src/storage/rowset/rowset_writer.cpp]: > 1. Add a tablet info in this error Status.

done, changed in https://github.com/StarRocks/starrocks/pull/17264/commits/8527758eaac88fd366e07ed4bd2cdf2b3f757dde

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by decster — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by decster** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Pull request merged**